react-components/src/metadata/admin/EditMetadataTemplate.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @author psarando
+ */
+import React, { Component } from "react";
+import { Field, FieldArray, reduxForm } from "redux-form";
+import { injectIntl } from "react-intl";
+
+import build from "../../util/DebugIDUtil";
+import withI18N, { formatMessage, getMessage } from "../../util/I18NWrapper";
+import withStoreProvider from "../../util/StoreProvider";
+import intlData from "../messages";
+import styles from "../style";
+import ids from "./ids";
+
+import ConfirmCloseDialog from "../../util/ConfirmCloseDialog";
+import { FormCheckbox, FormTextField } from "../../util/FormField";
+import EditAttributeFormList from "./EditAttributeFormList";
+import SlideUpTransition from "./SlideUpTransition";
+
+import AppBar from "@material-ui/core/AppBar";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import Divider from "@material-ui/core/Divider";
+import IconButton from "@material-ui/core/IconButton";
+import Toolbar from "@material-ui/core/Toolbar";
+import Typography from "@material-ui/core/Typography";
+import { withStyles } from "@material-ui/core/styles";
+
+import CloseIcon from "@material-ui/icons/Close";
+
+class EditMetadataTemplate extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = { showConfirmationDialog: false };
+    }
+
+    onSaveTemplate = ({ name, description, deleted, attributes }) => {
+        this.closeConfirmationDialog();
+
+        return new Promise((resolve, reject) => {
```

Comment:
Any reason for using `Promise` here instead just calling presenter with callbacks ?

Reply:
It's the only way [redux-form will set `subitting` to true](https://redux-form.com/7.3.0/docs/api/props.md/#-code-submitting-boolean-code-), which I use to disable the `Save` button to prevent duplicate submissions.

Reply:
👍